cg/meta/clean/clean_case_directories.py

```diff
@@ -0,0 +1,25 @@
+import logging
+from datetime import datetime
+from os import unlink
+from os.path import getmtime
+from pathlib import Path
+
+from cg.utils.date import get_date_days_ago
+
+LOG = logging.getLogger(__name__)
+
+
+def clean_directory(directory_to_clean: Path, days_old: int, dry_run: bool):
+    """Cleans the old /analysis/cases directories."""
+    date_threshold: datetime = get_date_days_ago(days_old)
+    for case_dir in directory_to_clean.iterdir():
+        case_modified_time: datetime = datetime.fromtimestamp(getmtime(case_dir))
+        if case_modified_time < date_threshold:
```

Comment:
I know that this is how we in general determine whether something can be removed, this approach seems really sketchy to me though.

I'm guessing there are several strict criteria which determine whether or not we actually can remove something but this module just assumes that those things have happened?

I'm guessing at least these criteria are assumed to have been fulfilled:

- The data has been uploaded.
- The data has been archived.

Ideally, our cleaning logic would explicitly make sure that these (and others I probably don't know about) are actually fulfilled.

Reply:
Or I guess in this case, the underlying assumption is that the analysis for the case has been cancelled?

Reply:
I was not a part of the refinement meeting but the instructions here https://github.com/Clinical-Genomics/cg/issues/2756#issuecomment-1885003308, showed no such assumptions.

Reply:
Any input @fevac ?

Reply:
Even clearer: https://github.com/Clinical-Genomics/MTP-RNAFUSION/issues/26#issuecomment-2120157249

Reply:
What it was previously decided was to remove directories if they were old at a first stage. Then in a different PR attempt to fix the issues with status not being consistent or tracked (which needs refinement).  But if there are new concerns we could have a new refinement session for this.

Reply:
I'd advice against blindly deleting data based on timestamps, if that is how this cleaning script is supposed to work.

Reply:
We can't assume that data has been persisted elsewhere just because it has not been modified in X days

Reply:
Changed the logic to not use the timestamp blindly